### PR TITLE
Issue #9531: update WriteTagCheckTest to use unique Input file

### LINKED ISSUE #9531: update WriteTagCheckTest to use unique Input file in each test method
WriteTagCheckTest.java file reuse some Input files few time in few different test method.
Expected: each test method in Test class use it own Input file and do not share it with any other test method.

more details - https://github.com/checkstyle/checkstyle/wiki/Good-practice-on-how-to-create-Input-files-for-tests

### PR DESCRIPTION
**Pull request by AkMo3**
Closes #9531

### CODE REVIEW

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/writetag/InputWriteTag.java [hunk 1]
```diff
@@ -6,30 +6,34 @@
 package com.puppycrawl.tools.checkstyle.checks.javadoc.writetag;
 
 /**
+ * Config:
+ * tag = &#64;author
+ * tagFormat = \\S
+ *
  * Testing tag writing
```
Comment (timurt) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/writetag/InputWriteTag.java:13]: is it possible to separate `Config` comments from actual example? it is a little bit confusing to understand :)

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/writetag/InputWriteTag.java [hunk 2]
```diff
@@ -6,30 +6,34 @@
...
-        /**
-         * @todo Add a constructor comment
-         */
-        public InputWriteTag()
-        {
-        }
+    /**
+     * @todo Add a constructor comment
+     */
+    public InputWriteTag() // ok
```
Comment (timurt) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/writetag/InputWriteTag.java:25]: you do not have to write trailing `// ok` comments, based on [these suggestions](https://github.com/checkstyle/checkstyle/wiki/Good-practice-on-how-to-create-Input-files-for-tests), only `violations`

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/writetag/InputWriteTag.java [hunk 3]
```diff
@@ -6,30 +6,34 @@
...
+ *
  * Testing tag writing
  * @author Daniel Grenner
  * @incomplete This class needs more code...
  * @doubletag first text
  * @doubletag second text
  * @emptytag
  */
-class InputWriteTag
+class InputWriteTag // violation at 14
```
Comment (timurt) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/writetag/InputWriteTag.java:20]: if violation occurred at 14, why comment is here? :)

**Review by timurt — CHANGES_REQUESTED** [review]

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/writetag/InputWriteTag.java (hunk 1 above)
Comment (AkMo3) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/writetag/InputWriteTag.java:15]: Done.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/writetag/InputWriteTag.java (hunk 3 above)
Comment (AkMo3) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/writetag/InputWriteTag.java]: I didn't wanted to amend the violation message with ` // violation` at the end. Well changes are done now.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/writetag/InputWriteTag.java (hunk 2 above)
Comment (AkMo3) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/writetag/InputWriteTag.java]: Sir, this is true, but what about files which have no violations. Also, we have been following this practice so I prefer to keep this. If you insist I would remove them, but for the time being I have kept them.

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/WriteTagCheckTest.java [hunk 4]
```diff
@@ -193,12 +193,12 @@ public void testEnumsAndAnnotations() throws Exception {
...
+            "22: " + getCheckMessage(MSG_WRITE_TAG, "@incomplete",
                 "This enum constant needs more code..."),
-            "19: " + getCheckMessage(MSG_WRITE_TAG, "@incomplete",
+            "28: " + getCheckMessage(MSG_WRITE_TAG, "@incomplete",
                 "This annotation needs more code..."),
-            "23: " + getCheckMessage(MSG_WRITE_TAG, "@incomplete",
+            "32: " + getCheckMessage(MSG_WRITE_TAG, "@incomplete",
                 "This annotation field needs more code..."),
         };
         verify(checkConfig, getPath("InputWriteTag2.java"), expected);
```
Comment (timurt) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/WriteTagCheckTest.java:204]: could you please rename it according to test case

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/WriteTagCheckTest.java [hunk 5]
```diff
@@ -208,7 +208,7 @@ public void testEnumsAndAnnotations() throws Exception {
     public void testNoJavadocs() throws Exception {
         final DefaultConfiguration checkConfig = createModuleConfig(WriteTagCheck.class);
         final String[] expected = {
-            "3: " + getCheckMessage(MSG_MISSING_TAG, "null"),
+            "4: " + getCheckMessage(MSG_MISSING_TAG, "null"),
         };
         verify(checkConfig, getPath("InputWriteTag3.java"), expected);
```
Comment (timurt) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/WriteTagCheckTest.java:213]: same here, rename it please to something related to test name

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/writetag/InputWriteTag2.java [hunk 6]
```diff
@@ -6,21 +6,31 @@
...
  */
-enum InputWriteTag2 {
+class InputWriteTag2 {
     /**
-     * @incomplete This enum constant needs more code...
-     */
-    FOO;
-}
+     * @incomplete This enum needs more code...
+     */ // violation
```
Comment (timurt) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/writetag/InputWriteTag2.java:19]: I have noticed you placed all `// violation` comments not on the same line where expected violation occurred, why so?

**Review by timurt — CHANGES_REQUESTED** [review]

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/writetag/InputWriteTag2.java (hunk 6 above)
Comment (AkMo3) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/writetag/InputWriteTag2.java]: I forget to commit change on this file. Only this file was left to do changes suggested previously. Done now. Please check.

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/WriteTagCheckTest.java (hunk 5 above)
Comment (AkMo3) [reply src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/WriteTagCheckTest.java]: Done

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/WriteTagCheckTest.java (hunk 4 above)
Comment (AkMo3) [reply src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/WriteTagCheckTest.java]: Done

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/writetag/InputWriteTag.java (hunk 3 above)
Comment (AkMo3) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/writetag/InputWriteTag.java:20]: Please resolve this one if everything is ok.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/writetag/InputWriteTag.java (hunk 2 above)
Comment (AkMo3) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/writetag/InputWriteTag.java:25]: Sir, please share your final views on this.

Comment (nmancus1) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/writetag/InputWriteTag.java]: We require one occurrence of ` * Config` and one occurrence of  `// ok` or `// violation` in all check input files:  https://github.com/checkstyle/checkstyle/blob/81df29c69f57579f247b76bafdf1393cc28a1997/config/checkstyle_input_checks.xml#L19

**Review by romani — APPROVED** [review]
@timurt , please finalize review

**Review by timurt — APPROVED** [review]

**Pull request merged**